quarkchain/cluster/tests/test_cluster.py

```diff
@@ -2430,3 +2451,91 @@ def mock_get_root_chain_stakes(recipient, _):
             # fail again, because quota used up
             with self.assertRaises(ValueError):
                 add_root_block(staker_addr, sign=True)
+
+    def test_grpc_slave(self):
+        id1 = Identity.create_random_identity()
+        acc1 = Address.create_from_identity(id1, full_shard_key=0)
+
+        with ClusterContext(1, acc1, connect_grpc=True) as clusters:
+            block_header_list = [clusters[0].get_shard_state(2 | 0).header_tip]
+            shard_state0 = clusters[0].get_shard_state(0b10)
+            for i in range(7):
+                b1 = _tip_gen(shard_state0)
+                add_result = call_async(
+                    clusters[0].master.add_raw_minor_block(
+                        b1.header.branch, b1.serialize()
+                    )
+                )
+                self.assertTrue(add_result)
+                block_header_list.append(b1.header)
+
+            block_header_list.append(clusters[0].get_shard_state(2 | 1).header_tip)
+            shard_state0 = clusters[0].get_shard_state(0b11)
+            b2 = _tip_gen(shard_state0)
+            add_result = call_async(
+                clusters[0].master.add_raw_minor_block(b2.header.branch, b2.serialize())
+            )
+            self.assertTrue(add_result)
+            block_header_list.append(b2.header)
+
+            root_block = clusters[0].master.root_state.create_block_to_mine(
+                block_header_list, acc1
+            )
+            # Test Case 1 ###################################################
+            # This case tests the correct connection
+            normalserver1 = NormalServer()
+            server1, port_number1 = self.build_test_server(
+                normalserver1,
+                clusters[0].master.env.cluster_config.GRPC_SLAVE_LIST[0].HOST,
+                clusters[0].master.env.cluster_config.GRPC_SLAVE_LIST[0].PORT,
+            )
+            server1.start()
+            server2, port_number2 = self.build_test_server(
+                normalserver1,
+                clusters[0].master.env.cluster_config.GRPC_SLAVE_LIST[1].HOST,
+                clusters[0].master.env.cluster_config.GRPC_SLAVE_LIST[1].PORT,
+            )
+            server2.start()
+            self.assertEqual(
```

Comment:
I don't think it's necessary to assert `port_number` since they are not part of the code we want to test